webapp/src/jaas/components/conference/jaas_conference.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import {id as pluginId} from '../../../manifest';
+import {StartMeetingWindowAction} from 'jaas/actions';
+
+type Props = {
+    actions: {
+        startJaaSMeetingWindow: (jwt: string | null, path: string | null) => Promise<StartMeetingWindowAction>
+    }
+}
+
+type State = {
+    jaasJwt?: string | null,
+    jaasRoom?: string | null,
+}
+
+const JWT = 'jwt';
+const JAAS_URL = '8x8.vc';
+
+export default class JaaSConference extends React.PureComponent<Props, State> {
+    api: any;
+    style = getStyle();
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {};
+    }
+
+    initJaaS(jwt: string, room: string) {
+        const url = new URL(window.location.href);
+        const noSSL = url.protocol === 'https:';
+
+        const ws = Math.max(document.documentElement.clientWidth ?? 0, window?.innerWidth ?? 0);
+        const hs = Math.max(document.documentElement.clientHeight ?? 0, window?.innerHeight ?? 0);
+        const options = {
+            width: ws,
+            height: hs,
+            roomName: room,
+            jwt,
+            noSSL,
+            parentNode: document.querySelector('#jitsiMeet')
+        };
+
+        this.api = new (window as any).JitsiMeetExternalAPI(JAAS_URL, options);
+    }
+
+    componentDidMount() {
+        if (!(window as any).JitsiMeetExternalAPI) {
+            const script = document.createElement('script');
+            script.type = 'text/javascript';
+            script.onload = () => {
+                if (!this.state.jaasRoom) {
+                    const params = new URLSearchParams(window.location.search);
+                    const jwt = params.get(JWT);
+                    const path = window.location.pathname;
+                    this.props.actions.startJaaSMeetingWindow(jwt, path).
+                        then((response) => {
+                            this.setState({
+                                jaasRoom: response.data.jaasRoom,
+                                jaasJwt: response.data.jaasJwt
+                            });
+                        });
+                }
+            };
+            script.src = `${(window as any).location.origin}/plugins/${pluginId}/jitsi_meet_external_api.js`;
+            document.head.appendChild(script);
```

Comment:
It seems like a bit of an antipattern to load this in a component's mount method. What are the side effects of this loading twice?

Reply:
A separate topic, what are the side effects of this loading twice? When the plugin is redeployed, it may run this script again. Maybe we should have some client-side state used to make sure we don't load the script twice.

Reply:
I am not able to find any side effects of loading this script twice. I think it's a bit overkill to have a redux state just to check whether this script is already loaded or not because redeployment of the plugin is not something we do multiple times.

Reply:
@Kshitij-Katiyar Can you test out what happens when you disable/re-enable the plugin in this case?

Reply:
@mickmister Please find the below video for these steps:- 

https://user-images.githubusercontent.com/90389917/206996147-33ea8329-97de-4762-a460-e3fdf85497d8.mp4